Tests/SourceKitLSPTests/DocumentTestDiscoveryTests.swift

```diff
@@ -418,6 +418,43 @@ final class DocumentTestDiscoveryTests: XCTestCase {
     )
   }
 
+  func testSwiftTestingTestsWithDuplicateFunctionIdentifiers() async throws {
```

Comment:
I think it would be good to also have a test that we disambiguate tests inside a suite.

```swift
struct MySuite {
  @Test(arguments: [1, 2, 3])
  func foo(_ x: Int) {}
  @Test(arguments: ["a", "b", "c"])
  func foo(_ x: String) {}
}
```

Or, slightly more complicated

```swift
struct MySuite {
  @Test(arguments: [1, 2, 3])
  func foo(_ x: Int) {}
}
extension MySuite {
  @Test(arguments: ["a", "b", "c"])
  func foo(_ x: String) {}
}
```

Reply:
Your intuition was right, duplicate IDs discovered in extensions was a corner case that wasn't covered by the implementation. I've updated to a simpler (but less efficient) two pass solution that addresses this.